racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -0,0 +1,153 @@
+<!--Copyright (c) 2022-2023 Rackslab
+
+This file is part of RacksDB.
+
+SPDX-License-Identifier: GPL-3.0-or-later -->
+
+<script setup lang="ts">
+import { useHttp } from '@/plugins/http'
+import { ref, onMounted, inject } from 'vue'
+import SearchBar from '@/components/SearchBar.vue'
+// import { injectionKey } from '@/plugins/runtimeConfiguration'
+import type { Ref } from 'vue'
+import type { Datacenter, Infrastructure, Rack } from '@/composables/RacksDBAPI'
+
+const datacenters: Ref<Array<Datacenter>> = ref([])
+const infrastructures: Ref<Array<Infrastructure>> = ref([])
+const datacenterDetails: Ref<Datacenter | undefined> = ref()
+const racks: Ref<Array<Rack>> = ref([])
+const rackDetails: Ref<Array<Rack>> = ref([])
+//const showFullImg = ref(false)
+const http = useHttp()
+
+/*
+function toggleImageModal() {
+  showFullImg.value = !showFullImg.value
+}
+*/
+
+// this function checks in which infrastructures the rack is present and returns the infrastructure name(s)
+function listInfrastructures() {
+  const rack = rackDetails.value
+  const infrastructure = infrastructures.value
+  const infrastructureNames = []
+
+  for (let index = 0; index < rack.length; index++) {
+    const rackName = rack[index].name
+
+    for (let x = 0; x < infrastructure.length; x++) {
+      for (let y = 0; y < infrastructure[x].layout.length; y++) {
+        const infrastructuresRacks = infrastructure[x].layout[y].rack
+
+        if (infrastructuresRacks == rackName) {
+          infrastructureNames.push(infrastructure[x].name)
+          return infrastructureNames
+        }
+      }
+    }
+  }
+}
+
+async function getInfrastructures() {
+  try {
+    const resp = await http.get('infrastructures')
+    infrastructures.value = resp.data as Infrastructure[]
+  } catch (error) {
+    console.error('Error during infrastructures data recovery', error)
+  }
+}
+
+async function getRacks() {
+  try {
+    const resp = await http.get('racks')
+    racks.value = resp.data as Rack[]
+    rackDetails.value = racks.value.filter((rack) => rack.room === props.datacenterRoom)
+  } catch (error) {
+    console.error('Error during racks data recovery', error)
+  }
+}
+
+async function getDatacenters() {
+  try {
+    const resp = await http.get('datacenters')
+    datacenters.value = resp.data as Datacenter[]
+    datacenterDetails.value = datacenters.value.filter(
+      (datacenter) => datacenter.name === props.datacenterName
+    )[0]
+  } catch (error) {
+    console.error('Error during datacenters data recovery', error)
+  }
+}
+
+onMounted(() => {
+  getDatacenters()
+  getInfrastructures()
+  getRacks()
+})
+
+const props = defineProps({
+  datacenterName: String,
+  datacenterRoom: String
+})
+</script>
+
+<template>
+  <SearchBar
+    v-if="datacenters.length"
+    viewTitle="Datacenter Room"
+    searchedItem="datacenter"
+    :items="datacenters"
+  />
+
+  <h2 class="flex justify-center py-10 capitalize text-3xl">{{ datacenterRoom }} room</h2>
+
+  <!--
+  <img
```

Comment:
To display room and infrastructure images, you will need to request RacksDB API with Axios and POST method. You will probably need to specify that the type of the response data is a blob. Then you will be able to create a fictional URL with `URL.URL.createObjectURL(blob)` and use this generated URL in `src` attribute of `<img />` tag.